api/v1beta1/azuremachine_types.go

```diff
@@ -132,6 +132,12 @@ type AzureMachineSpec struct {
 	// +optional
 	DNSServers []string `json:"dnsServers,omitempty"`
 
+	// DisableBootstrappingVMExtension disables CAPZ Bootstrapping VM extension.
+	// Its role is to detect and report Kubernetes bootstrap failure or success.
+	// Use this setting only if VMExtensions are not supported by your image.
+	// +optional
+	DisableBootstrappingVMExtension *bool `json:"disableBootstrappingVMExtension,omitempty"`
```

Comment:
@RadekManak sorry for the confusion, I was following this effort from the edge but didn't make the effort to comment. I actually prefered the previous "disable all VM extensions" option. I think that is a more clarifying option to fulfill your problem, and would help other users in a similar scenario where VM extensions are not supported.

Disabling this particular extension doesn't make a lot of sense as you lose an explicit part of the CAPI bootstrapping contract guarantee; but it *does* make sense if your CAPZ VM scenario doesn't allow extensions that we want to ship a VM configuration with no Extensions in the configuration.

IMO

We'll prioritize merging this change w/ the previous implementation, thank you so much for this and for your patience!